packages/grafana-ui/src/types/datasource.ts

```diff
@@ -106,6 +106,11 @@ export interface DataSourceApi<TQuery extends DataQuery = DataQuery> {
    */
   getQueryHints?(query: TQuery, results: any[], ...rest: any): QueryHint[];
 
+  /**
+   * Convert a query to a simple text string
+   */
+  getCollapsedText?(query: TQuery): string;
```

Comment:
* getCollapsedQueryText(query)    feels more specific

Reply:
I'll change it -- I just copied from the angular version :)

Reply:
What is the difference between query's text and collapsed text? Not sure if the `collapsed` adjective is not blurring the essence of this method.

Reply:
Well it's a shortform, "display" form, might not show everything, only essentials. For data sources that have non string based query languages like elasticsearch  it just becomes more of a render essential options to a formatted string.

Reply:
OK, then it makes sense. Since it was confusing I suggest adding an example in the comment above.

Reply:
I like `getQueryDisplayText`

Reply:
done, and passed CI.  

Still something funny with my permissions that I can't merge things after they are approved.  This is what I see:
![image](https://user-images.githubusercontent.com/705951/55894670-62499e00-5b6f-11e9-9456-3d9ee1c901cb.png)